response_operations_ui/static/js/read-csv.js

```diff
@@ -1,88 +1,91 @@
+
+        document.getElementById('sample-preview').innerHTML = preview;
+        document.getElementById('btn-check-sample-contents').style.display = 'none';
+        document.getElementById('btn-load-sample').style.display = 'inline-block';
+        document.getElementById('btn-cancel-load-sample').style.display = 'inline-block';
+    }
 
-function drawOutput(businessCount, ciCount){
-
-	//Clear previous data
-	document.getElementById("sample-preview").innerHTML = "";
-
-	var preview = "";
-	preview = preview + "<div class='panel panel--info'>";
-	preview = preview + "  <div class='panel__header'>";
-	preview = preview + "    <h3 class='u-fs-r--b'>Sample contents</h3>";
-	preview = preview + "  </div>";
-	preview = preview + "  <div class='panel__body'>";
-	preview = preview + "    <div id='sample-preview-businesses'>Number of businesses: " + businessCount + "</div>";
-	preview = preview + "    <div id='sample-preview-ci'>Collection instruments: " + ciCount + "</div>";
-	preview = preview + "  </div>";
-	preview = preview + "</div>";
-
-	document.getElementById("sample-preview").innerHTML = preview;
-	document.getElementById("btn-check-sample-contents").style.display = "none";
-	document.getElementById("btn-load-sample").style.display = "inline-block";
-	document.getElementById("btn-cancel-load-sample").style.display = "inline-block";
-}
-
-function errorHandler(evt) {
-	if(evt.target.error.name == "NotReadableError") {
-		alert("Cannot read file :(");
-	}
-}
-
-function handleFiles(files, classifiers) {
-	// Check for the various File API support.
-	if (window.FileReader) {
-		// FileReader are supported.
-		var reader = new FileReader();
-
-        // Handle errors load
-        reader.onload = function(evt) {
-            processFile(evt, classifiers);
-        };
-        reader.onerror = errorHandler;
+    function errorHandler(evt) {
+        if (evt.target.error.name == 'NotReadableError') {
+            alerts.error('Error Uploading file', 'The selected file was not readable.');
+        }
+    }
 
-        // Read file into memory as UTF-8
-        reader.readAsText(files[0]);
+    function handleFiles(files, classifiers) {
+        if (_browserHasFileLoaderCapability()) {
+            const reader = new FileReader();
 
-	} else {
-		alert('FileReader is not supported in this browser.');
-	}
-}
+            reader.onload = function(evt) {
+                processFile(evt, classifiers);
+            };
 
-function processFile(event, classifiers) {
-    var csv = event.target.result;
-    var allTextLines = csv.split(/\r\n|\n/);
-    var classifierColumn = [];
-    var lines = [];
-    var ciCount;
+            reader.onerror = errorHandler;
 
-    while (allTextLines.length) {
-        var line = allTextLines.shift().split(":")
-        if (line && line.length && line[0]) {
-            lines.push(line);
+            reader.readAsText(files[0]);
+        } else {
+            // @TODO: Implement better way to handle this
```

Comment:
I think this comment should either be expanded to say how it could/should be fixed, or just removed.  I've never liked having TODOs in code, especially ones that don't give a hint on how to do the thing that needs to be done.
Maybe make a trello card too so that it doesn't get forgotten about?

Reply:
Heh, ironically, I've actually fixed what this TODO referred to, and omitted to remove the todo.

Fix on the way.